src/Raven.Studio/typescript/viewmodels/database/query/query.ts

```diff
@@ -313,13 +314,15 @@ class query extends viewModelBase {
             this.queriedIndex() ? appUrl.forVisualizer(this.activeDatabase(), this.queriedIndex()) : null);
 
         this.isMapReduceIndex = ko.pureComputed(() => {
-            const currentIndex = this.getCurrentIndex();
+            this.getQueriedIndexInfo();
+            const currentIndex = this.queriedIndexInfo();
             return !!currentIndex && (currentIndex.Type === "AutoMapReduce" || currentIndex.Type === "MapReduce" || currentIndex.Type === "JavaScriptMapReduce");
         });
 
         this.queryResultsContainMatchingDocuments = ko.pureComputed(() => {
-            const currentIndex = this.getCurrentIndex();
-            return !!currentIndex && currentIndex.SourceType !== "TimeSeries" && currentIndex.SourceType !== "Counters" && !this.isMapReduceIndex();
+            this.getQueriedIndexInfo();
```

Comment:
we should call methods with side effects inside pure computed. as it will be invoked on every change.